typing/btype.ml

```diff
@@ -100,19 +101,19 @@ let rec field_kind_repr =
     Fvar {contents = Some kind} -> field_kind_repr kind
   | kind                        -> kind
 
-let rec repr_link compress t d =
+let rec repr_link compress (t : type_expr) d : type_expr -> type_expr =
  function
    {desc = Tlink t' as d'} ->
      repr_link true t d' t'
  | {desc = Tfield (_, k, _, t') as d'} when field_kind_repr k = Fabsent ->
      repr_link true t d' t'
  | t' ->
      if compress then begin
-       log_change (Ccompress (t, t.desc, d)); t.desc <- d
+       log_change (Ccompress (t, t.desc, d)); (Internal.unlock t).desc <- d
```

Comment:
There are a few lines like this. Maybe add an `Internal.set_desc` to make them easier to read.

Reply:
The idea here is that one should not directly modify types, so providing a function that does it in `types.mli` seems a mixed message. Or change the name `Internal` to something more scary like `Unsafe_access`.